pyop2/caching.py

```diff
@@ -240,14 +241,109 @@ def cache_key(self):
 .. note::
     If you intend to use this decorator to cache things that are collective
     across a communicator then you must include the communicator as part of
-    the cache key. Since communicators are themselves not hashable you should
-    use :func:`pyop2.mpi.hash_comm`.
+    the cache key.
 
     You should also make sure to use unbounded caches as otherwise some ranks
     may evict results leading to deadlocks.
 """
 
 
+def default_parallel_hashkey(*args, **kwargs):
+    comm = kwargs.get('comm')
+    return comm, cachetools.keys.hashkey(*args, **kwargs)
+
+
+def parallel_memory_only_cache(key=default_parallel_hashkey):
```

Comment:
Bit of a mouthful, `parallel_cache` instead? then could have `parallel_disk_cache` if needed.